designs/karpenter-integration-testing.md

```diff
@@ -0,0 +1,271 @@
+# Karpenter - Integration Testing RFC
+
+Note: This document serves as a living specification for Karpenter's integration testing strategy and orchestration mechanisms. As the project evolves, both the design requirements and this documentation will be updated to reflect current best practices and project needs.
+
+## Summary
+
+The AWS Karpenter Provider currently serves as the primary testing ground for Karpenter's core functionality, with comprehensive integration tests covering provisioning, scheduling, disruption handling, networking configurations, and AWS-specific feature implementations. In the kubernetes-sigs/karpenter (Core) repository, testing is more limited, primarily utilizing the KWOK (Kubernetes Without Kubelet) Provider with KWOKNodeClass for node simulation to validate basic provisioning and deprovisioning functionality. Each Karpenter Providers would currently need to need to maintain a set smiler set of integration tests to validate the functionality of any updates to the core project, creating testing duplication across the community. This is particularly painful as each cloud provider would need to be aware of functionality changes and create/update their integration tests 
+
+The testing framework for the AWS Karpenter Provider has several types of tests:
+
+* Unit Tests
+    * Run in a mock Kubernetes environment
+    * Part of the regular developer workflow
+
+* Integration Tests
+    * Create real clusters and interact with actual APIs
+    * Validate the cluster changes 
+    * Run on every commit
+* Scale Tests
+    * Will validate Karpenter's performance at large scales
+    * Focus on horizontal scaling workloads and resource management
```

Comment:
What resources are we talking about here? Resources provisioned by Karpenter as NodeClaims or resources consumed by the Karpenter controller pod?

Reply:
This would be resources provisioned by Karpenter. Updated to reflect that